core/templates/pages/exploration-editor-page/editor-tab/templates/modal-templates/customize-interaction-modal.component.ts

```diff
@@ -254,10 +256,32 @@ export class CustomizeInteractionModalComponent
   }
 
   isSaveInteractionButtonEnabled(): boolean {
-    return !!(
-      this.hasCustomizationArgs &&
-      this.stateInteractionIdService.displayed &&
-      (this.getCustomizationArgsWarningsList().length === 0));
+    let explorationTitle = this.getTitle(
+      this.stateInteractionIdService.displayed);
+    let EndExploration = 'End Exploration';
+    /*
+      There is a condition for the END EXPLORATION interaction
+      We have to check that the input field for adding
+      the exploration Ids must be non-empty
+      Save Interaction button should be disable for empty input
+    */
+    if (explorationTitle === EndExploration) {
+      let inputField = document.getElementsByClassName('exploration-id-input');
+      if (inputField === undefined) {
+        return true;
+      }
+      let inputValue = angular.element(
+        inputField[inputField.length - 1]).attr('ng-reflect-model');
```

Comment:
What does this line do? Why are you fetching `inputField[inputField.length - 1]`?

Reply:
I am checking that if the user has clicked Add Exploration ID, then the last input box must not be empty. If so then save interaction button must be disabled! Also if any other input box other than the last one is empty, then the button should be enabled as the empty box other than the last one gets automatically deleted if empty.

Reply:
I understand.
But, would this not fail if the length of the `inputField` is 0?

Also, I'm not quite convinced that this is the solution to the problem. The problem is that even after some change in the exploration IDs list, the save button stays disabled. The core issue to this problem is that the cust arg warnings aren't getting updated when a change is made to the cust arg since that change hasn't been saved yet so the warning is still present and `this.getCustomizationArgsWarningsList()` length is > 0 which is causing this function to return false. Your workaround to just check if the last input is empty or not seems unrelated and the reason it's working is that you've basically overridden the condition that was catching the warnings.

The ideal solution would be to re-evaluate the cust arg warnings in real-time while this cust arg is being edited. But that might not be viable (a backend call on every change would be expensive) or even completely necessary.

One hacky way to do this could be to explicitly ignore this particular invalid IDs warning during the `this.getCustomizationArgsWarningsList().length === 0` check. This should be safe since the cust arg warnings are displayed outside anyways.

@Shivkant-Chauhan Please see if there's a better solution.

cc: @Rijuta-s

Reply:
done as per discussed!